thorn/include/kernel/mailbox.h

```diff
@@ -32,9 +32,12 @@ typedef enum {
 typedef struct {
     unsigned int data    : 28;// most significant bits contain shared memory address
     channel_t    channel : 4; // least four significant bits contain channel
-} mbox_message_t;
+} mailbox_message_t;
 
 // Send message and check responses
-bool mailbox_request (volatile unsigned int * data_ptr, channel_t channel);
+bool         mailbox_request (volatile unsigned int * buffer, channel_t channel);
+unsigned int mailbox_message_to_register_value (mailbox_message_t * message);
```

Comment:
```suggestion
unsigned int mailbox_message_to_register_value (mailbox_message_t message);
```
`mailbox_message_t` is 4 bytes long, there's no need to pass it as pointer. Generally, as long as structs are 8 bytes maximum, you can safely pass them by value - after all, the pointer you'd pass otherwise is 8 bytes.

Passing by value ensures no modifications happen and, since the compiler can reason about this and doesn't need another indirection, is likely to be faster.

If you do want to pass by reference, you should probably at least make it `mailbox_message const * message`, as promise that you don't modify it.

The same goes for `mailbox_read` and `mailbox_write`; none of these modify `message`